tasks/main.yml

```diff
@@ -1 +1,94 @@
-# role tasks
+# tasks file for azure-driver
+- import_role:
+    name: oasis_roles.system_repositories
+  vars:
+    system_repositories_configs:
+      - name: azure-cli
+        baseurl: https://packages.microsoft.com/yumrepos/azure-cli
+        gpgcheck: true
+        gpgkey: https://packages.microsoft.com/keys/microsoft.asc
+
+- name: Install Azure CLI
+  package:
+    name: azure-cli
+    state: present
+
+- name: Install jq
+  package:
+    name: jq
+    state: present
+
+- name: Ensure Azure credential folder exists
+  file:
+    path: "~/.azure/"
+    state: directory
+
+# These json files allow us to skip the az login interactive step.
+- name: Copy json files to ~/.azure
+  copy:
+    src: "{{ lookup('env', 'WORKSPACE') }}\
+/scenario/css_openshift/config/ansible/roles/azure_infra/files/{{ item }}"
+    dest: "~/.azure/{{ item }}"
+  with_items:
+    - azureProfile.json
+    - accessTokens.json
+
+# We can't guarantee the SP will be the same, so let's pull a new one every run
+# and update the credentials file
+- name: Get new service principal secret
+  command: "az ad sp create-for-rbac --role Contributor \
+            --name {{ azure_sp_name }}"
+  register: command_result
+  changed_when: command_result.rc == 0
```

Comment:
`changed_when` is for clarifying to ansible when it should report that a task made a change. Ansible automatically fails a command task if the `rc` is nonzero, so this `changed_when` is never considered in cases where the `changed_when` condition would be false. There are several instances of `changed_when` in this role's tasks, and as far as I can tell they should all be removed.

I assume this is to avoid the "command shouldn't change anything when nothing needs doing" error out of ansible lint, but the correct way to solve that is by either doing some inspection to make sure a command really needs to be run before running it (i.e. add a `when` clause), or to explicitly make no attempt at idempotence and declare the lack of idempotence to role users in the role README.